daemon/algod/api/server/v2/utils.go

```diff
@@ -184,7 +184,7 @@ func getCodecHandle(formatPtr *string) (codec.Handle, string, error) {
 
 	switch format {
 	case "json":
-		return protocol.JSONHandle, "application/json", nil
+		return protocol.JSONStrictHandle, "application/json", nil
```

Comment:
is this really needed for this PR ? I'm worried that it might cause some regression elsewhere. i.e.
```golang
// JSONStrictHandle is the same as JSONHandle but with MapKeyAsString=true
// for correct maps[int]interface{} encoding
var JSONStrictHandle *codec.JsonHandle
```

Reply:
Without this change the test becomes more complicated because `NUM_TRANSACTIONS=$(cat "${TEMPDIR}/curl_out.txt" | jq '.block.txns | length')` doesn't work.

Reply:
I just thought of an alternative and have removed this change.

That said, we need to use JSONStrictHandle. The API should not respond with invalid JSON.